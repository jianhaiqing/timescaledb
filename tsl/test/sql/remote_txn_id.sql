-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

\c :TEST_DBNAME :ROLE_SUPERUSER
CREATE FUNCTION _timescaledb_internal.test_remote_txn_id()
RETURNS void
AS :TSL_MODULE_PATHNAME, 'tsl_test_remote_txn_id'
LANGUAGE C STRICT;

SELECT _timescaledb_internal.test_remote_txn_id();

SELECT '1-10-20'::rxid;

create table tbl_w_rxid(
    txn_id rxid
);

CREATE UNIQUE INDEX idx_name ON tbl_w_rxid ((txn_id::text));

INSERT INTO tbl_w_rxid VALUES ('1-10-20'), ('1-11-20'), ('1-10-21');

SELECT txn_id, _timescaledb_internal.rxid_in(_timescaledb_internal.rxid_out(txn_id))::text = txn_id::text FROM tbl_w_rxid;

\set ON_ERROR_STOP 0
INSERT INTO tbl_w_rxid VALUES ('1-10-20');

SELECT ''::rxid;
SELECT '--'::rxid;
SELECT '1-10-20a'::rxid;
SELECT '2-10-20'::rxid;
SELECT '1-10-20.0'::rxid;
SELECT '1-10.0-20'::rxid;
SELECT 'a1-10-20'::rxid;
