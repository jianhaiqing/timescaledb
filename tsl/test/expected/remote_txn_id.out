-- Copyright (c) 2016-2018  Timescale, Inc. All Rights Reserved.
--
-- This file is licensed under the Timescale License,
-- see LICENSE-TIMESCALE at the top of the tsl directory.
\c :TEST_DBNAME :ROLE_SUPERUSER
CREATE FUNCTION _timescaledb_internal.test_remote_txn_id()
RETURNS void
AS :TSL_MODULE_PATHNAME, 'tsl_test_remote_txn_id'
LANGUAGE C STRICT;
SELECT _timescaledb_internal.test_remote_txn_id();
 test_remote_txn_id 
--------------------
 
(1 row)

SELECT '1-10-20'::rxid;
  rxid   
---------
 1-10-20
(1 row)

create table tbl_w_rxid(
    txn_id rxid
);
CREATE UNIQUE INDEX idx_name ON tbl_w_rxid ((txn_id::text));
INSERT INTO tbl_w_rxid VALUES ('1-10-20'), ('1-11-20'), ('1-10-21');
SELECT txn_id, _timescaledb_internal.rxid_in(_timescaledb_internal.rxid_out(txn_id))::text = txn_id::text FROM tbl_w_rxid;
 txn_id  | ?column? 
---------+----------
 1-10-20 | t
 1-11-20 | t
 1-10-21 | t
(3 rows)

\set ON_ERROR_STOP 0
INSERT INTO tbl_w_rxid VALUES ('1-10-20');
ERROR:  duplicate key value violates unique constraint "idx_name"
SELECT ''::rxid;
ERROR:  invalid input syntax for remote transaction ID: '' at character 8
SELECT '--'::rxid;
ERROR:  invalid input syntax for remote transaction ID: '--' at character 8
SELECT '1-10-20a'::rxid;
ERROR:  invalid input syntax for remote transaction ID: '1-10-20a' at character 8
SELECT '2-10-20'::rxid;
ERROR:  invalid version for remote transaction ID: 2 at character 8
SELECT '1-10-20.0'::rxid;
ERROR:  invalid input syntax for remote transaction ID: '1-10-20.0' at character 8
SELECT '1-10.0-20'::rxid;
ERROR:  invalid input syntax for remote transaction ID: '1-10.0-20' at character 8
SELECT 'a1-10-20'::rxid;
ERROR:  invalid input syntax for remote transaction ID: 'a1-10-20' at character 8
